src/mpi/request/waitall.c

```diff
@@ -67,57 +67,25 @@ static inline int request_complete_fastpath(MPI_Request *request, MPIR_Request *
 #define FUNCNAME MPIR_Waitall_impl
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-int MPIR_Waitall_impl(int count, MPI_Request array_of_requests[],
+int MPIR_Waitall_impl(int count, MPIR_Request *request_ptrs[],
                       MPI_Status array_of_statuses[])
 {
     int mpi_errno = MPI_SUCCESS;
-    MPIR_Request * request_ptr_array[MPIR_REQUEST_PTR_ARRAY_SIZE];
-    MPIR_Request ** request_ptrs = request_ptr_array;
-    MPI_Status * status_ptr = NULL;
+    MPI_Status * status_ptr;
     MPID_Progress_state progress_state;
-    int i, j;
+    int i;
     int n_completed;
-    int active_flag;
     int rc = MPI_SUCCESS;
     int n_greqs;
-    int proc_failure = FALSE;
     int disabled_anysource = FALSE;
     const int ignoring_statuses = (array_of_statuses == MPI_STATUSES_IGNORE);
-    int optimize = ignoring_statuses; /* see NOTE-O1 */
-    MPIR_CHKLMEM_DECL(1);
-
-    /* Convert MPI request handles to a request object pointers */
-    if (count > MPIR_REQUEST_PTR_ARRAY_SIZE)
-    {
-        MPIR_CHKLMEM_MALLOC(request_ptrs, MPIR_Request **, count * sizeof(MPIR_Request *), mpi_errno, "request pointers", MPL_MEM_OBJECT);
-    }
 
     n_greqs = 0;
     n_completed = 0;
     for (i = 0; i < count; i++)
     {
-	if (array_of_requests[i] != MPI_REQUEST_NULL)
+        if (request_ptrs[i] != NULL)
```

Comment:
nitpick: indent is off by one? (I wouldn't worry too much about the style as we are going to apply the code cleanup script soon anyway.)